catalog/spatial/registry/registry-report-viewer/src/main/java/org.codice.ddf.registry.report.viewer/RegistryReportViewer.java

```diff
@@ -0,0 +1,399 @@
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.PostalAddressType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.RegistryObjectListType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.RegistryObjectType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.RegistryPackageType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.ServiceBindingType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.ServiceType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.SlotType1;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.TelephoneNumberType;
+
+@Path("/")
+public class RegistryReportViewer {
+    private static final Logger LOGGER = LoggerFactory.getLogger(RegistryReportViewer.class);
+
+    private String registryName = "";
+
+    private static final String BINDING_TYPE = "bindingType";
+
+    private FederationAdminService federationAdminService;
+
+    private ClassPathTemplateLoader templateLoader;
+
+    public RegistryReportViewer() {
+        templateLoader = new ClassPathTemplateLoader();
+        templateLoader.setPrefix("/templates");
+        templateLoader.setSuffix(".hbt");
+    }
+
+    @Path("/{metacardId}/report")
+    @Produces(MediaType.TEXT_HTML)
+    @GET
+    public Response viewRegInfoHtml(@PathParam("metacardId") final String metacardId,
+            @QueryParam("sourceId") final List<String> sourceIds) {
+        Handlebars handlebars = new Handlebars(templateLoader);
+        String html = "";
+        if (StringUtils.isBlank(metacardId)) {
+            return Response.status(Response.Status.BAD_REQUEST)
+                    .entity("Metacard Id cannot be blank")
+                    .build();
+        }
+
+        RegistryPackageType registryPackage;
+        try {
+            registryPackage = federationAdminService.getRegistryObjectByMetacardId(metacardId,
+                    sourceIds);
+        } catch (FederationAdminException e) {
+            String message = "Error getting registry package.";
+            LOGGER.error("{} For metacard id: '{}', optional sources: {}",
+                    message,
+                    metacardId,
+                    sourceIds);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR)
+                    .entity(message)
+                    .build();
+        }
+
+        if (registryPackage == null) {
+            String message = "No registry package was found.";
+            LOGGER.error("{} For metacard id: '{}', optional source ids: {}.",
+                    message,
+                    metacardId,
+                    sourceIds);
+            return Response.status(Response.Status.NOT_FOUND)
+                    .entity(message)
+                    .build();
+        }
+
+        Map<String, Object> registryMap = buildRegistryMap(registryPackage);
+
+        try {
+            Template template = handlebars.compile("report");
+            html = template.apply(registryMap);
+        } catch (IOException e) {
+        }
```

Comment:
How does this sound? `LOGGER.error("Error compiling and applying report template.");`